controllers/contacts.js

```diff
@@ -0,0 +1,79 @@
+const fs = require("fs").promises;
+
+const path = require("path");
+const contactsPath = path.resolve("models/contacts.json");
+
+const listContacts = async () => {
+  const data = await fs.readFile(contactsPath, "utf-8");
+  return JSON.parse(data);
+};
+
+const getContactById = async (id) => {
+  const data = await fs.readFile(contactsPath);
+  const parseData = JSON.parse(data);
+  return parseData.find((contact) => contact.id === id);
+};
+
+const removeContact = async (contactId) => {
+  try {
+    const data = await fs.readFile(contactsPath);
+    const filterContacts = JSON.parse(data).filter(
+      (data) => Number(data.id) !== Number(contactId)
+    );
+    await fs.writeFile(contactsPath, JSON.stringify(filterContacts));
+    return listContacts();
+  } catch (err) {
+    return err;
+  }
+};
+
+const addContact = async (name, email, phone) => {
+  try {
+    const data = await fs.readFile(contactsPath);
+    const dataParse = JSON.parse(data);
```

Comment:
nie ma potrzeby za każdym razem definiowania dwóch zmiennych aby otrzymać zparsowane dane z pliku 

używając poniżego kodu osiągniemy ten sam efekt

```suggestion
  const data = await fs.readFile(contactsPath, 'utf-8');
```